src/anm/player_manager.js

```diff
@@ -40,4 +45,23 @@ PlayerManager.prototype.getPlayer = function(cvs_id) {
     return this.hash[cvs_id];
 };
 
-module.exports = new PlayerManager();
+PlayerManager.prototype.handleDocumentHiddenChange = function(hidden) {
+    var i, player;
+    for(i=0;i<this.instances.length;i++) {
+        player = this.instances[i];
+        if (hidden && player.state.happens === C.PLAYING) {
+            player._pausedViaHidden = true;
+            player.pause();
+        } else if (!hidden && player._pausedViaHidden) {
+            player._pausedViaHidden = false;
+            player.play(player.state.from);
+        }
+    }
+};
+
+var manager = new PlayerManager();
+engine.onDocumentHiddenChange = function(hidden) {
```

Comment:
I think modifying something in the engine from outside is not a very good way, could it be refactored like this?:

`engine.onDocumentHiddenChange(function(hidden) { manager.handleDocumentHiddenChange(hidden); });`, which in its turn calls `addEventListener` with corresponding event, if it's there, or does nothing if it is not supported?